feat(instances): Fix broken VPN when switching openvpn instance to ro… (#3825)
scaleway/docs-content: compute/instances/troubleshooting/fix-broken-vpn-when-switching-openvpn-vm-to-routed-ip.mdx

---
meta:
  title: Fix broken VPN when switching OpenVPN Instance to routed IP

>   description: This page helps you recover VPN functionality when switching your OpenVPN instance to routed IP

Reviewer: ```suggestion
  description: This page helps you recover VPN functionality when switching your OpenVPN Instance to routed IP
```

content:

>   h1: Fix broken VPN when switching openvpn instance to routed IP

Reviewer 2: ```suggestion
  h1: Fix broken VPN when switching OpenVPN Instance to routed IP
```

>   paragraph: This page helps you recover VPN functionality when switching your OpenVPN instance to routed IP

Reviewer: ```suggestion
  paragraph: This page helps you recover VPN functionality when switching your OpenVPN Instance to routed IP
```

tags: openvpn  instance routed ip
dates:
  validation: 2024-10-11
  posted: 2024-10-11
categories:
  - compute
---

> When switching an OpenVPN Instance to routed IP, the configured VPN will fail to work after the reboot. This is caused by a bug in a script that has been fixed on newer versions of the OpenVPN image.

Reviewer: ```suggestion
When switching an Instance created with the OpenVPN InstantApp to routed IP, the configured VPN will fail to work after the reboot. This is caused by a bug in a script that has been fixed on newer versions of the OpenVPN image.
```

> This can now be fixed by installing a new version of the OpenVPN scripts which has been packaged Scaleway's stable PPA that contained the fixed script.

Reviewer: There's a preposition missing here somewhere but I can't work out where. "into which has been packaged"? or "which contains a package for Scaleway's stable bla bla"?

Reviewer 2: ```suggestion
This can now be fixed by installing a new version of the OpenVPN scripts, which has been packaged in Scaleway's stable PPA and contains the fixed script.
```

> You can also avoid this situation alltogether by installing the package prior to moving to a routed IP.

Reviewer: ```suggestion
You can also avoid this situation altogether by installing the package prior to moving to a routed IP.
```

## Your Instance has not been rebooted

1. Add Scaleway's stable PPA

```sh
add-apt-repository ppa:scaleway/stable
```
2. Add the ovpn-scripts package containing the new scripts

```sh
apt -y install ovpn-scripts
```

From this point on, your Instance may be safely rebooted and you will no longer lose VPN functionality.

## Your Instance has rebooted and the VPN service no longer works.

1. Add Scaleway's stable PPA

```sh
add-apt-repository ppa:scaleway/stable
```
2. Add the ovpn-scripts package containing the new scripts

```sh
apt -y install ovpn-scripts
```

> 3. Reboot your instance

Reviewer 2: ```suggestion
3. Reboot your Instance.
```
